pyingest/parsers/ads_nameparser.py

```diff
@@ -0,0 +1,191 @@
+import sys
+import re
+from namedentities import named_entities
+
+try:
+    import ads.ADSCachedExports as ads_ex
+except ImportError:
+    sys.path.append('/proj/ads/soft/python/lib/site-packages')
+    try:
+        import ads.ADSCachedExports as ads_ex
+    except ImportError as e:
+        print 'Unable to import ads libraries: {}'.format(e)
+
+
+import pyingest.config.config as config
+from adsputils import u2asc
+
+RE_INITIAL = re.compile('\. *(?!,)')
+etal = re.compile(r",? et ?al\.?")
+reand = re.compile(r" and ")
+redash = re.compile(r"^-")
+requot = re.compile(r"^'")
+
+first_names = []
+last_names = []
+
+
+def read_datfile(filename):
+    output_list = []
+    with open(filename,'rU') as fp:
+        for l in fp.readlines():
+            if l.strip() != '' and l[0] != '#':
+                output_list.append(l.strip())
+    return output_list
+
+
+def init_namelists():
```

Comment:
This is not efficient as written.  Right now we have `ads_name_adjust()` call `init_namelists()` each time it's executed, and `init_namelists()` reads four files every time it's executed!  I suggest you define these as module level variables (or even class-level if you want to go the OO way).

Reply:
Fixed -- loaded at the module level now.